R/sqlOnly.R

```diff
@@ -0,0 +1,221 @@
+  )
+  
+  check_query <- SqlRender::loadRenderTranslateSql(
+    sqlFilename = file.path("sqlOnly", "cte_sql_for_results_table.sql"),
+    packageName = "DataQualityDashboard",
+    dbms = connectionDetails$dbms,
+    query_text = gsub(";", "", sql),
+    check_name = checkDescription$checkName,
+    check_level = checkDescription$checkLevel,
+    check_description = gsub("'", "''", check_description),
+    cdm_table_name = check["cdmTableName"],
+    cdm_field_name = check["cdmFieldName"],
+    concept_id = check["conceptId"],
+    unit_concept_id = check["unitConceptId"],
+    sql_file = checkDescription$sqlFile,
+    category = checkDescription$kahnCategory,
+    subcategory = checkDescription$kahnSubcategory,
+    context = checkDescription$kahnContext,
+    checkid = .getCheckId(checkDescription$checkLevel, checkDescription$checkName, check["cdmTableName"], check["cdmFieldName"], check["conceptId"], check["unitConceptId"]),
+    threshold_value = thresholdValue,
+    query_num = qnum
+  )
+
+  sql_to_union <<- append(sql_to_union, check_query)
+}
+
+
+#' Internal function to write queries when running in SqlOnly mode
+#' 
+#' @noRd
+#' @keywords internal
+#' 
+.writeSqlOnlyQueries <- function(
+  sql_to_union,
+  sqlOnlyUnionCount,
+  resultsDatabaseSchema,
+  writeTableName,
+  dbms,
+  outputFolder,
+  checkDescription
+) {
+  outFile <-file.path(
+    outputFolder, 
+    sprintf("%s_%s.sql", checkDescription$checkLevel, checkDescription$checkName)
+  )
+
+  # Delete existing file
+  unlink(outFile)
+
+  ustart <- 1
+  while (ustart <= length(sql_to_union)) {
+    uend <- min(ustart + sqlOnlyUnionCount - 1, length(sql_to_union))
+    
+    sql_unioned <- paste(sql_to_union[ustart:uend], collapse=' UNION ALL ')
+    
+    sql <- SqlRender::loadRenderTranslateSql(
+      sqlFilename = file.path("sqlOnly", "insert_ctes_into_result_table.sql"),
+      packageName = "DataQualityDashboard",
+      dbms = dbms,
+      resultsDatabaseSchema = resultsDatabaseSchema,
+      tableName = writeTableName,
+      query_text = sql_unioned
+    )
+    
+    write(
+      x = sql,
+      file = outFile,
+      append = TRUE
+    )
+    
+    ustart <- ustart + sqlOnlyUnionCount
+  }
+}
+
+
+#' Internal function to write the DDL to outputFolder
+#' @noRd
+#' @keywords internal
+.writeDDL <- function(
+    resultsDatabaseSchema,
+    writeTableName,
+    dbms,
+    outputFolder
+) {
+  tableName <- sprintf("%s.%s", resultsDatabaseSchema, writeTableName)
+
+  sql <- SqlRender::loadRenderTranslateSql(
+    sqlFilename = "result_dataframe_ddl.sql",
+    packageName = "DataQualityDashboard",
+    dbms = dbms,
+    tableName = tableName
+  )
+  
+  write(
+    x = sql,
+    file = file.path(
+      outputFolder, 
+      "ddlDqdResults.sql"
+    )
+  )
+}
+
+
+#' Internal function to evaluate one threshold
+#' Note: this does not evaluate is_error or not_applicable status
+#' @noRd
+#' @keywords internal
+.evaluateOneThresholdSqlOnly <- function(
```

Comment:
should this be named "get threshold" instead, since it's not actually evaluating the threshold?

Reply:
good idea - will do